src/LoginControl.test.js

```diff
@@ -23,28 +17,50 @@ beforeEach(() => {
 });
 
 it("renders without crashing", () => {
-  shallow(<LoginControl />);
+  mockIsLoggedIn.mockReturnValueOnce(false);
+  shallow(<LoginControl isLoggedIn={mockIsLoggedIn}/>);
 });
 
-it("renders correctly (snapshot testing)", () => {
-  const wrapper = mount(<LoginControl />);
+it("renders LoginButton when isLoggedIn returns false (snapshot testing)", () => {
+  mockIsLoggedIn.mockReturnValueOnce(false);
+  const wrapper = mount(<LoginControl isLoggedIn={mockIsLoggedIn} />);
+  expect(wrapper).toMatchSnapshot();
+
+  wrapper.unmount();
+});
+
+it("renders LogoutButton when isLoggedIn returns true (snapshot testing)", () => {
+  mockIsLoggedIn.mockReturnValueOnce(true);
+  const wrapper = mount(<LoginControl isLoggedIn={mockIsLoggedIn} />);
   expect(wrapper).toMatchSnapshot();
 
   wrapper.unmount();
 });
 
```

Comment:
spec doesn't contain the cases when isLoggedIn is not present in the props or when it is not a function.